### ISSUE TITLE
v2.3.4 of NetBox adds an extra backslash ("/") between hostname and URL 

### ISSUE THREAD

**Issue by bdlamprecht** [digitalocean/pynetbox#61]
### Issue type
[X] Bug report      

### Environment
* Python version:  3.5
* NetBox version:  2.3.4
* PyNetBox version: 3.3.1

### Description
Using `pynetbox` querying against an instance of `NetBox` running `v2.3.3` works fine.
However, when I switch the URL to an instance of `NetBox` running `v2.3.4`, it stops working.

The details and code for each version are below (the same exact data is in both versions of `NetBox`).

---

(`v2.3.3`)
```
>>> try:
...     nb.dcim.sites.get(slug='0am')
... except pynetbox.lib.query.RequestError as e:
...     print(e.req.url)
...
0am
```

---

(`v2.3.4`)
```
>>> try:
...     nb.dcim.sites.get(slug='0am')
... except pynetbox.lib.query.RequestError as e:
...     print(e)
...
The request failed with code 404 Not Found
>>> try:
...     nb.dcim.sites.get(slug='0am')
... except pynetbox.lib.query.RequestError as e:
...     print(e.req.url)
...
http://myhost:8080//api/dcim/sites/?slug=0am
```

---

Notice the extra `/` between the hostname and the rest of the URL?
I'm close to positive that is what is causing the issue.
However, when using the `SwaggerUI` and performing the same action via a `PUT` request, no issues.
So I'm not sure if the error exists in `NetBox` since that is what changed or `pynetbox` since everything **else** in `v2.3.4` works fine.

**Comment by konaya** [comment]
I can confirm this.

`Jun 25 09:54:51 dcim gunicorn[13044]: b'' - - [25/Jun/2018:07:54:51 +0000] "GET //api/dcim/devices/125/ HTTP/1.0" 404 13580 "-" "python-requests/2.10.0"`

Removing the extra slash (it's a forward slash, by the way – a backslash is a "\") made it work as predicted:

`Jun 25 09:55:58 dcim gunicorn[13044]: b'' - - [25/Jun/2018:07:55:58 +0000] "GET /api/dcim/devices/125/ HTTP/1.0" 200 1986 "-" "python-requests/2.10.0"`

**Comment by konaya** [comment]
I'm not sure if I'd call it a bug, though. The example in README.md of pynetbox clearly shows that an ending slash shouldn't be included in the URL. Sure, it happened to work with multiple slashes in succession before, but that is by no means a defined behaviour.

Either make the documentation more explicit in this regard, make pynetbox autodetect and discard any extraneous slashes, or convince the Netbox project to update their API spec to allow for extra slashes. The former alternatives being the better ones, IMHO.

**Comment by zachmoody** [comment]
I think pynetbox has always raised a RequestError when passed a trailing slash. I've thought about sanitizing the url, but haven't gotten around to it. I tested it on `2.3.3` and `2.3.4` and got it to raise and exception on each, but I also couldn't reproduce the issue. 🤷‍♂️

**Comment by bdlamprecht** [comment]
So I'm not sure what I'm doing wrong and how @konaya was able to
> Remove the extra slash

and not encounter the error with `v2.3.4`.

I'm not manually add a `/` anywhere, the `/` gets added automatically by `pynetbox` when I do a `nb.dcim.sites.get(slug='0am')`

**Comment by zachmoody** [comment]
I'm not sure that pynetbox is adding it. I've tested with the docker image on docker-hub and couldn't reproduce the behavior.

**Comment by bdlamprecht** [comment]
Ah, I think I found what I was doing incorrectly.
For my original example for this issue, I was using the following code:
```
import pynetbox,requests

token     = '0123456789abcdef0123456789abcdef01234567'
test_url  = 'http://node2.internal:10123/'
node_url  = 'http://node3.internal:10123/'

nb_url    = test_url
nb        = pynetbox.api(nb_url,token)
```
This worked fine for all versions of `NetBox` through `v2.3.3`, however, in `v2.3.4`, I had to remove the trailing _**forward**_ slash (`/`) from the end of the `url`:
```
import pynetbox,requests

token     = '0123456789abcdef0123456789abcdef01234567'
test_url  = 'http://node2.internal:10123'
node_url  = 'http://node3.internal:10123'

nb_url    = test_url
nb        = pynetbox.api(nb_url,token)
```
After I made that change, everything started to work fine once again.

Like @konaya said, the README.md of `pynetbox` does show the `url` without the trailing slash, however I don't know that I would call it **_clearly_** :)

@zachmoody I think it would be a good idea to either, (1) sanitize it like you mentioned wanting to or (2) updating the documentation to make it clear that including the trailing `/` might cause problems.
